docs/usage.md

```diff
@@ -79,43 +83,58 @@ See also `cachi2 generate-env --help`.
 
 ### Inject project files
 
+While some package managers only need an environment file to be informed of the cache locations, others may need to
+create a configuration file or edit a lockfile (or some other file in your project directory).
+
+Before starting your build, call `cachi2 inject-files` to automatically make the necessary changes in your repository
+(based on data in the fetch-deps output directory). Please do not change the absolute path to the repo between the calls
+to fetch-deps and inject-files; if it's not at the same path, the inject-files command won't find it.
+
 ```shell
 cachi2 inject-files ./cachi2-output --for-output-dir /tmp/cachi2-output
 ```
 
+The `--for-output-dir` option has the same meaning as the one used when generating environment variables.
+
 *⚠ Cachi2 may overwrite existing files. Please make sure you have no un-committed changes (that you are not prepared to
 lose) when calling inject-files.*
 
-For some package managers, to use the pre-fetched dependencies you may need to create a configuration file or edit
-a lockfile (or some other file in your project directory).
+*⚠ Cachi2 may change files if required by the package manager. This means that the git status will become dirty if
+it was previously clean. If any scripting depends on the cleanliness of a git repository and you do not want to commit
+the changes, the scripting should either be changed to handle the dirty status or the changes should be temporarily
+stashed by wrapping in `git stash && <command> && git stash pop` according to the suitability of the context.*
 
-Before starting your build, call `cachi2 inject-files` to automatically make the necessary changes in your repo (based
-on data in the fetch-deps output directory). Please do not change the absolute path to the repo between the calls to
-fetch-deps and inject-files; if it's not at the same path, the inject-files command won't find it.
+### Building the Artifact with the Pre-fetched dependencies
 
-The `--for-output-dir` option has the same meaning as the one used when generating environment variables.
+After the pre-fetch and the above steps to inform the package manager(s) of the cache have been completed, it all
+needs to be wired up into a build. The primary use case for building these is within a Dockerfile or Containerfile
+but the same principles can be applied to other build strategies.
 
 ### Write the Dockerfile (or Containerfile)
 
+Now that we have pre-fetched our dependencies and enabled package manager configuration to point to them, we now need
+to ensure that the build process (i.e. a Dockerfile or Containerfile for a container build) is properly written to
+build in a network isolated mode. All injected dependencies are changed in the source itself, so they will be present
```

Comment:
Ah, now I understand. But to someone else, it may not be clear if "injected dependencies" is about the dependencies downloaded by `fetch-deps` or about the files injected by `inject-files`

Maybe "injected files"? Not sure if it's any better, but it's a little closer to the terminology used earlier